src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -12852,10 +12852,9 @@ static void Main()
 
             comp.VerifyIL("C.Main", @"
 {
-  // Code size       86 (0x56)
-  .maxstack  3
-  .locals init (System.ValueTuple<int, int> V_0,
-                System.ValueTuple<long, object> V_1)
+  // Code size       65 (0x41)
```

Comment:
Notice the IL change. This is now classified an explicit tuple literal conversion, and 
since it is a literal conversion (not a conversion from a tuple type) we do not materialize an intermediate tuple and IL is more compact.

There is a subtle difference in the order of sideeffects between evaluation of the tuple arguments and element-wise conversions. We are now consistent with implicit tuple literal conversions here and element conversions are interlaced with argument evaluations. There is no intermediate tuple materialization between argument eval and element conversions. 

We did discuss this at LDM, if you wonder... it is a subtle change though.